lib/passport-saml/saml.js

```diff
@@ -837,7 +837,9 @@ SAML.prototype.hasValidSignatureForRedirect = function (container, originalQuery
 
 function validateSignatureForRedirect (urlString, signature, alg, cert) {
   var supportedAlgs = crypto.getHashes().filter(
-    function(h) { return new RegExp(h).test(alg); }
+    function(h) { return new RegExp(h, 'i').test(alg); }
+  ).sort(
+    function(h) { return -h.length; }
```

Comment:
Why are we even testing with a RegExp here? Wouldn't it be safer and less error-prone to test with an exact match instead? I can see possibly testing case-insensitive.   

But if the algorithm used was `mdc2`, shouldn't we use the `mdc2` algorithm and not the `mdc2WithRSA` algorithm?

The way I read this code, it tests with a case-insensitive substring match and then chooses the algorithm with the longest name that matches. 

We also throw away the entire array but the last entry after running a regex over over hash and then sorting it. 

It would be clearer and faster to stop looking as soon as find the match we we want. 

We could skip using `supportAlgs` at all. The current code was using a case-sensitive match and it seemed to work fine. If we test for exact matches, then the incoming `alg` can be passed through to `createVerify()` assuming it's an algorithm supported by Node.js

Reply:
the current version was also taking the last entry:
```js
function validateSignatureForRedirect (urlString, signature, alg, cert) {
  var supportedAlgs = crypto.getHashes().filter(
    function(h) { return new RegExp(h).test(alg); }
  );
  if (supportedAlgs.length === 0) {
    throw alg + ' is not supported';
  }
  var verifier = crypto.createVerify(supportedAlgs[supportedAlgs.length-1]);
  verifier.update(urlString);
  return verifier.verify(cert, signature, 'base64');
}
```